lib/dogma/rules/final_newline.ex

```diff
@@ -8,7 +8,9 @@ defmodule Dogma.Rules.FinalNewline do
   alias Dogma.Script
   alias Dogma.Error
 
-  def test(script) do
+  def test(script), do: test(script, [])
+
+  def test(script, _config = []) do
```

Comment:
For final_newline (and in fact most of the rules) we don't have any config that can be provided. So I'd say calling it with anything else other than an empty config list is an error.